scripts/patch.js

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env node
+'use strict'
+
+/*
+ * Patch the items in the local releases dir into the current dist.ipfs.io
+ * For each item
+ *   - if it does not exist in dist, add it.
+ *   - if it exists and has the same cid, skip it.
+ *   - if it exists and is a file, update it.
+ *   - it it exists and is a dir, recurse.
+ *
+ * Builds an array of operations required to patch the local files into distRoot,
+ * then applies those changes.
+ *
+ * Prints the CID of the patched dist site, for pinning to cluster  and publishing.
+ */
+
+const fs = require('fs')
+const path = require('path')
+const all = require('it-all')
+const chalk = require('chalk')
+const IpfsHttpClient = require('ipfs-http-client')
+const { globSource } = IpfsHttpClient
+require('make-promises-safe') // exit on error
+
+const ipfs = IpfsHttpClient()
+const pathTo = (file) => path.join(__dirname, '..', file)
+
+const DIST_DOMAIN = 'dist.ipfs.io'
```

Comment:
nit: I'd consider using a random string `dist-.....` until we're done so we don't accidentally clobber something (and so we know the process "completed").

Reply:
I think it is necessary to use the real domain, since that is resolved to the current distribution, and then used to calculate what needs to be patched.

Reply:
Ah, sorry, I put the comment in the wrong location. I meant use a random string for MFS_DIR.

Reply:
In case I, e.g., ran a partial build and failed somewhere along the way.

Reply:
Appended ISO data string to `dist.ipfs.io`, to be unique plus indicate when dist was published.
```
> ipfs files ls
dist.ipfs.io_2021-02-01T21:50:30.407Z
```

Reply:
this is fine and good. I am compelled to link to the fledgling "cohosting" spec which has an opinion on MFS dir names for snapshots of dnslink'd websites https://github.com/ipfs-shipyard/cohosting/blob/master/SPEC.md#path-conventions. It's only mildly relevant here as the goal of this script is not co-hosting the dist site, but is worth knowing about.